src/core/thread/indirect_sender.cpp

```diff
@@ -418,37 +418,56 @@ uint16_t IndirectSender::PrepareDataFrame(Mac::TxFrame &aFrame, Child &aChild, M
 
 void IndirectSender::PrepareEmptyFrame(Mac::TxFrame &aFrame, Child &aChild, bool aAckRequest)
 {
-    uint16_t     fcf;
-    Mac::Address macSource, macDest;
-
+    Mac::Address macDest;
     aChild.GetMacAddress(macDest);
+    PrepareEmptyFrame(aFrame, macDest, aAckRequest);
+}
 
+void IndirectSender::PrepareEmptyFrame(Mac::TxFrame &aFrame, Mac::Address &aMacDest, bool aAckRequest)
+{
+    uint16_t fcf = 0;
+
+    Mac::Address macSource;
     macSource.SetShort(Get<Mac::Mac>().GetShortAddress());
 
-    if (macSource.IsShortAddrInvalid() || macDest.IsExtended())
+    if (macSource.IsShortAddrInvalid() || aMacDest.IsExtended())
     {
         macSource.SetExtended(Get<Mac::Mac>().GetExtAddress());
     }
 
-    fcf = Mac::Frame::kFcfFrameData | Mac::Frame::kFcfFrameVersion2006 | Mac::Frame::kFcfPanidCompression |
-          Mac::Frame::kFcfSecurityEnabled;
+    fcf = Mac::Frame::kFcfFrameData | Mac::Frame::kFcfPanidCompression | Mac::Frame::kFcfSecurityEnabled;
 
     if (aAckRequest)
     {
         fcf |= Mac::Frame::kFcfAckRequest;
     }
 
-    fcf |= (macDest.IsShort()) ? Mac::Frame::kFcfDstAddrShort : Mac::Frame::kFcfDstAddrExt;
+    fcf |= (aMacDest.IsShort()) ? Mac::Frame::kFcfDstAddrShort : Mac::Frame::kFcfDstAddrExt;
     fcf |= (macSource.IsShort()) ? Mac::Frame::kFcfSrcAddrShort : Mac::Frame::kFcfSrcAddrExt;
+    {
+        FILE *p = fopen("debug.log", "a+");
+        fprintf(p, "dest short:%d, src short:%d\n", aMacDest.IsShort(), macSource.IsShort());
+        fclose(p);
+    }
```

Comment:
```
    {
         FILE *p = fopen("debug.log", "a+");
         fprintf(p, "dest short:%d, src short:%d\n", aMacDest.IsShort(), macSource.IsShort());
         fclose(p);
     }
```

What is intention behind this block? 
(is it added for testing only - if so please remove or use debug log instead)